roles/deploy_flask_app/tasks/setup_infra.yaml

```diff
@@ -0,0 +1,95 @@
+---
+- name: Describe bastion instance
+  amazon.aws.ec2_instance_info:
+    instance_ids:
+      - "{{ deploy_flask_app_bastion_instance_id }}"
+  register: bastion_info
+
+- name: Set common variables to be used later
+  ansible.builtin.set_fact:
+    deploy_flask_app__resource_prefix: "{{ bastion_info.instances.0.public_dns_name | split('.') | first }}"
+    deploy_flask_app__group_id: "{{ bastion_info.instances.0.security_groups[0].group_id }}"
+    deploy_flask_app__vm_image_id: "{{ bastion_info.instances.0.image_id }}"
+    deploy_flask_app__bastion_public_ip: "{{ bastion_info.instances.0.public_ip_address }}"
+    deploy_flask_app__public_subnet_id: "{{ bastion_info.instances.0.subnet_id }}"
+
+- name: Add bastion host to inventory
+  ansible.builtin.add_host:
+    hostname: bastion
+    ansible_ssh_user: "{{ deploy_flask_app_bastion_host_username }}"
+    ansible_host: "{{ deploy_flask_app__bastion_public_ip }}"
+    ansible_python_interpreter: auto
+    ansible_ssh_common_args: '-o "UserKnownHostsFile=/dev/null" -o StrictHostKeyChecking=no -i {{ deploy_flask_app_bastion_ssh_private_key_path }}'
+
+- name: Set variable for key pair
+  ansible.builtin.set_fact:
+    deploy_flask_app__workers_keypair_name: "{{ deploy_flask_app__resource_prefix }}-key"
+
+- name: Create key pair to connect to the workers
+  amazon.aws.ec2_key:
+    name: "{{ deploy_flask_app__workers_keypair_name }}"
+  register: keypair_result
+
+- name: Save key pair content into file on bastion host
```

Comment:
While running the webapp_ha_aurora.yaml playbook with the create operation, this task encounters a permission issue, causing it to fail.

```
fatal: [localhost -> bastion]: UNREACHABLE! => {
    "changed": false,
    "msg": "Failed to connect to the host via ssh: Warning: Permanently added '54.183.152.54' (ED25519) to the list of known hosts.\r\nfedora@54.183.152.54: Permission denied (publickey,gssapi-with-mic).",
    "unreachable": true
}
```